s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -572,16 +573,32 @@ import scala.util.{Failure, Success, Try}
   private def poolSettings(implicit settings: S3Settings, system: ActorSystem) =
     settings.forwardProxy.map(proxy => {
       val address = InetSocketAddress.createUnresolved(proxy.host, proxy.port)
-      val transport = proxy.credentials.fold(ClientTransport.httpsProxy(address))(
-        c => ClientTransport.httpsProxy(address, BasicHttpCredentials(c.username, c.password))
-      )
+      val transport: ClientTransport = proxy.scheme match {
+        case "https" =>
+          proxy.credentials.fold(ClientTransport.httpsProxy(address))(
+            c => ClientTransport.httpsProxy(address, BasicHttpCredentials(c.username, c.password))
+          )
+        case "http" =>
+          TCPTransport(address)
+      }
+      ConnectionPoolSettings(system).withConnectionSettings(ClientConnectionSettings(system).withTransport(transport))
+    })
 
-      ConnectionPoolSettings(system)
-        .withConnectionSettings(
-          ClientConnectionSettings(system)
-            .withTransport(transport)
+  private case class TCPTransport(address: InetSocketAddress) extends ClientTransport {
```

Comment:
Maybe rename to `ProxyTransport` or `ChangeTargetEndpointTransport` or something.